manifests/instance/reserve_port.pp

```diff
@@ -0,0 +1,9 @@
+# **NOTE: THIS IS A [PRIVATE](https://github.com/puppetlabs/puppetlabs-stdlib#assert_private) DEFINED TYPE**
+#
+# This is a 'canary' defined type that allow us to fail a compile in the case
+# that the `stunnel::interface` and `stunnel::connection` defined types have an
+# overlapping listen port.
+#
+define stunnel::instance::reserve_port {
```

Comment:
In this case, we're actually reserving the port *in puppet*, not on the system. It's to make Puppet fail, not to do something on the system.

I can rename it to whatever but it won't really make any more sense unless I go for the ultimate `reserver_port_in_puppet_has_nothing_to_do_with_the_system` naming scheme. It's hidden, so I could technically do that I suppose.

Reply:
It would only be hidden until a user sees the duplicate resource declaration error this is designed to throw. I think that is only slightly better than the stunnel service failing to start over and over again.